src/block/block_ext.c

```diff
@@ -414,14 +414,42 @@ __wt_block_off_remove_overlap(
     if (before != NULL && before->off + before->size > off) {
         WT_RET(__block_off_remove(session, block, el, before->off, &ext));
 
+        WT_ASSERT_WITH_ERR_MSG(session, ext->off + ext->size >= off + size, EINVAL,
```

Comment:
I don't think it's necessary to add this new macro. A `WT_ASSERT` will suffice. The assert will still print the expression that fails which should provide enough information.